src/covid_model_seiir_pipeline/marshall.py

```diff
@@ -0,0 +1,442 @@
+are fully deprecated and removed.
+"""
+from collections import defaultdict
+from contextlib import contextmanager
+import io
+from pathlib import Path
+import typing
+import zipfile
+
+import h5py
+import numpy
+import pandas
+
+from covid_model_seiir_pipeline.paths import (
+    Paths,
+    DRAW_FILE_TEMPLATE,
+)
+from covid_shared.shell_tools import mkdir
+
+
+class DataTypes:
+    """
+    Enumerations of data types as understood by the Marshall interface.
+    """
+    beta_scales = "beta_scales"
+    coefficient = "coefficients"
+    components = "components"
+    date = "dates"
+    fit_beta = "betas"  # TODO: no longer needed - fit is not a stage
+    parameter = "parameters"
+    regression_beta = "beta"
+
+    # types which serialize to a DataFrame
+    DataFrame_types = frozenset([
+        fit_beta,
+        parameter,
+        date,
+        coefficient,
+        regression_beta,
+        beta_scales,
+        components,
+    ])
+
+
+class Keys:
+    """
+    Identifies data payload types and stores related information.
+
+    Callers should never use init and instead use one of the many factory
+    methods which embed important related information to the variable
+    draw_id/location_id argument.
+    """
+    def __init__(self, data_type, template, **key_args):
+        self.data_type = data_type
+        self.template = template
+        self.key_args = key_args
+
+    # factories related to regression
+    @classmethod
+    def coefficient(cls, draw_id):
+        return cls(DataTypes.coefficient, DRAW_FILE_TEMPLATE, draw_id=draw_id)
+
+    @classmethod
+    def date(cls, draw_id):
+        return cls(DataTypes.date, DRAW_FILE_TEMPLATE, draw_id=draw_id)
+
+    @classmethod
+    def fit_beta(cls, draw_id):
+        return cls(DataTypes.fit_beta, DRAW_FILE_TEMPLATE, draw_id=draw_id)
+
+    @classmethod
+    def parameter(cls, draw_id):
+        return cls(DataTypes.parameter, DRAW_FILE_TEMPLATE, draw_id=draw_id)
+
+    @classmethod
+    def regression_beta(cls, draw_id):
+        return cls(DataTypes.regression_beta, DRAW_FILE_TEMPLATE, draw_id=draw_id)
+
+    # factories related to forecasting
+    @classmethod
+    def beta_scales(cls, scenario, draw_id):
+        return cls(DataTypes.beta_scales, DRAW_FILE_TEMPLATE, scenario=scenario, draw_id=draw_id)
+
+    @classmethod
+    def components(cls, scenario, draw_id):
+        return cls(DataTypes.components, DRAW_FILE_TEMPLATE, scenario=scenario, draw_id=draw_id)
+
+    # other methods/properties
+    @property
+    def key(self):
+        return self.template.format(**self.key_args)
+
+    @property
+    def seed(self):
+        """
+        Returns a seed value for enabling concurrency.
+
+        This is a sort of hack - we embed the concurrency story into the Keys
+        class which can then inform other things.
+        """
+        assert list(self.key_args) == ['draw_id'], 'TODO - expand Keys.seed'
+        return "draw-{draw_id}".format(**self.key_args)
```

Comment:
This `seed` is designed to help with naming files - something that the Marshall isn't expected to do.

In this case, since all data is saved by-draw we can get safe writes across different per-draw jobs by embedding the draw number in the filename.